fix(logging): incorrect FPS logging

TNeutron opened the pull request:
## Description of Change
The code provides fixes incorrect FPS calculation issue. The last_frame variable needs to be updated on each iteration, Immediately after computing the FPS of previous.

Before this commit - the last frame was being updated only once at the beginning of stream_handler function. The fps and avg_frame_time computation were completely wrong. This commit fixes the computation.

This commit - updates last_frame  variable on each iteration. results is correct fps calculation.

## Tests scenarios
I have tested my Pull Request on latest Arduino-esp32 core with ESP32-CAM (CAMERA_MODEL_AI_THINKER) with Core Debug Level - Verbose

## Related links
[Please provide links to related issue, PRs etc.](https://github.com/espressif/arduino-esp32/issues/10920)

(*eg. Closes #number of issue*)
Closes 1 issue ( #10920  )

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/espressif/arduino-esp32?pullRequest=10921) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/espressif/arduino-esp32?pullRequest=10921) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/arduino-esp32?pullRequest=10921) it.</sub>

me-no-dev commented:
Hi @TNeutron . I see the problem, but what you suggest is not the solution. `last_frame = fr_end;` should be added after the frame time is calculated by `int64_t frame_time = fr_end - last_frame;` in order to reset the time. Line 284 should become 
```cpp
int64_t frame_time = fr_end - last_frame;
last_frame = fr_end;
//...
```

TNeutron commented:
> Hi @TNeutron . I see the problem, but what you suggest is not the solution. `last_frame = fr_end;` should be added after the frame time is calculated by `int64_t frame_time = fr_end - last_frame;` in order to reset the time. Line 284 should become
> 
> ```c++
> int64_t frame_time = fr_end - last_frame;
> last_frame = fr_end;
> //...
> ```

You are correct. I have tested it and verify it's working. Log samples given below -
```10:23:04.509 -> [ 55924][I][app_httpd.cpp:302] stream_handler(): MJPG: 5035B 132ms (7.58fps), AVG: 29ms (34.48fps), Temperature: 58.33
10:23:04.543 -> [ 55949][I][app_httpd.cpp:302] stream_handler(): MJPG: 3915B 24ms (41.67fps), AVG: 29ms (34.48fps), Temperature: 58.33
10:23:04.577 -> [ 55962][I][app_httpd.cpp:302] stream_handler(): MJPG: 5035B 13ms (76.92fps), AVG: 27ms (37.04fps), Temperature: 58.33
10:23:04.577 -> [ 55989][I][app_httpd.cpp:302] stream_handler(): MJPG: 5811B 27ms (37.04fps), AVG: 28ms (35.71fps), Temperature: 58.33
10:23:04.662 -> [ 56041][I][app_httpd.cpp:302] stream_handler(): MJPG: 5811B 51ms (19.61fps), AVG: 29ms (34.48fps), Temperature: 58.89
10:23:04.662 -> [ 56054][I][app_httpd.cpp:302] stream_handler(): MJPG: 4073B 12ms (83.33fps), AVG: 28ms (35.71fps), Temperature: 58.89
10:23:04.703 -> [ 56078][I][app_httpd.cpp:302] stream_handler(): MJPG: 4684B 24ms (41.67fps), AVG: 29ms (34.48fps), Temperature: 58.33
10:23:04.703 -> [ 56098][I][app_httpd.cpp:302] stream_handler(): MJPG: 4684B 19ms (52.63fps), AVG: 29ms (34.48fps), Temperature: 58.89
10:23:04.703 -> [ 56118][I][app_httpd.cpp:302] stream_handler(): MJPG: 4684B 20ms (50.00fps), AVG: 29ms (34.48fps), Temperature: 58.33
10:23:04.739 -> [ 56138][I][app_httpd.cpp:302] stream_handler(): MJPG: 5003B 19ms (52.63fps), AVG: 28ms (35.71fps), Temperature: 58.89
10:23:04.776 -> [ 56158][I][app_httpd.cpp:302] stream_handler(): MJPG: 5086B 20ms (50.00fps), AVG: 28ms (35.71fps), Temperature: 58.33
10:23:04.776 -> [ 56191][I][app_httpd.cpp:302] stream_handler(): MJPG: 5811B 33ms (30.30fps), AVG: 29ms (34.48fps), Temperature: 58.89
10:23:04.808 -> [ 56204][I][app_httpd.cpp:302] stream_handler(): MJPG: 4626B 12ms (83.33fps), AVG: 29ms (34.48fps), Temperature: 58.33
10:23:04.808 -> [ 56218][I][app_httpd.cpp:302] stream_handler(): MJPG: 4866B 13ms (76.92fps), AVG: 29ms (34.48fps), Temperature: 58.33
10:23:04.847 -> [ 56238][I][app_httpd.cpp:302] stream_handler(): MJPG: 4866B 20ms (50.00fps), AVG: 29ms (34.48fps), Temperature: 58.89
10:23:04.880 -> [ 56268][I][app_httpd.cpp:302] stream_handler(): MJPG: 5811B 30ms (33.33fps), AVG: 30ms (33.33fps), Temperature: 58.89
10:23:04.880 -> [ 56292][I][app_httpd.cpp:302] stream_handler(): MJPG: 5811B 23ms (43.48fps), AVG: 30ms (33.33fps), Temperature: 58.89
10:23:04.913 -> [ 56306][I][app_httpd.cpp:302] stream_handler(): MJPG: 4444B 13ms (76.92fps), AVG: 28ms (35.71fps), Temperature: 58.89
10:23:04.913 -> [ 56322][I][app_httpd.cpp:302] stream_handler(): MJPG: 4920B 16ms (62.50fps), AVG: 28ms (35.71fps), Temperature: 59.44
10:23:04.945 -> [ 56351][I][app_httpd.cpp:302] stream_handler(): MJPG: 8090B 29ms (34.48fps), AVG: 27ms (37.04fps), Temperature: 58.89
10:23:04.985 -> [ 56368][I][app_httpd.cpp:302] stream_handler(): MJPG: 4860B 16ms (62.50fps), AVG: 21ms (47.62fps), Temperature: 58.33
10:23:04.985 -> [ 56381][I][app_httpd.cpp:302] stream_handler(): MJPG: 4860B 13ms (76.92fps), AVG: 21ms (47.62fps), Temperature: 58.33```

Please let me know if you'd like me to edit the commit. Thank you.

TNeutron commented:
> Hi @TNeutron . I see the problem, but what you suggest is not the solution. `last_frame = fr_end;` should be added after the frame time is calculated by `int64_t frame_time = fr_end - last_frame;` in order to reset the time. Line 284 should become
> 
> ```c++
> int64_t frame_time = fr_end - last_frame;
> last_frame = fr_end;
> //...
> ```

@me-no-dev 
You are correct. I have tested & verify it working. 
Log samples given below -

```
10:38:06.039 -> [  8154][I][app_httpd.cpp:303] stream_handler(): MJPG: 5338B 12ms (83.33fps), AVG: 210ms (4.76fps)
10:38:06.080 -> [  8177][I][app_httpd.cpp:303] stream_handler(): MJPG: 5376B 23ms (43.48fps), AVG: 209ms (4.78fps)
10:38:06.122 -> [  8214][I][app_httpd.cpp:303] stream_handler(): MJPG: 7149B 37ms (27.03fps), AVG: 209ms (4.78fps)
10:38:06.122 -> [  8237][I][app_httpd.cpp:303] stream_handler(): MJPG: 5881B 22ms (45.45fps), AVG: 209ms (4.78fps)
10:38:06.157 -> [  8255][I][app_httpd.cpp:303] stream_handler(): MJPG: 5042B 18ms (55.56fps), AVG: 209ms (4.78fps)
10:38:06.189 -> [  8283][I][app_httpd.cpp:303] stream_handler(): MJPG: 5881B 27ms (37.04fps), AVG: 209ms (4.78fps)
10:38:06.189 -> [  8304][I][app_httpd.cpp:303] stream_handler(): MJPG: 6046B 21ms (47.62fps), AVG: 208ms (4.81fps)
10:38:06.223 -> [  8337][I][app_httpd.cpp:303] stream_handler(): MJPG: 6046B 33ms (30.30fps), AVG: 206ms (4.85fps)
10:38:06.264 -> [  8356][I][app_httpd.cpp:303] stream_handler(): MJPG: 6046B 18ms (55.56fps), AVG: 204ms (4.90fps)
10:38:06.298 -> [  8384][I][app_httpd.cpp:303] stream_handler(): MJPG: 6046B 27ms (37.04fps), AVG: 205ms (4.88fps)
10:38:06.298 -> [  8404][I][app_httpd.cpp:303] stream_handler(): MJPG: 6046B 20ms (50.00fps), AVG: 204ms (4.90fps)
10:38:06.336 -> [  8425][I][app_httpd.cpp:303] stream_handler(): MJPG: 7149B 20ms (50.00fps), AVG: 204ms (4.90fps)
10:38:06.336 -> [  8451][I][app_httpd.cpp:303] stream_handler(): MJPG: 5519B 26ms (38.46fps), AVG: 204ms (4.90fps)
10:38:06.406 -> [  8502][I][app_httpd.cpp:303] stream_handler(): MJPG: 7149B 51ms (19.61fps), AVG: 206ms (4.85fps)
10:38:06.406 -> [  8515][I][app_httpd.cpp:303] stream_handler(): MJPG: 5297B 12ms (83.33fps), AVG: 205ms (4.88fps)
10:38:06.439 -> [  8551][I][app_httpd.cpp:303] stream_handler(): MJPG: 7149B 36ms (27.78fps), AVG: 205ms (4.88fps)
10:38:06.479 -> [  8571][I][app_httpd.cpp:303] stream_handler(): MJPG: 7149B 19ms (52.63fps), AVG: 205ms (4.88fps)
10:38:06.513 -> [  8596][I][app_httpd.cpp:303] stream_handler(): MJPG: 7149B 24ms (41.67fps), AVG: 206ms (4.85fps)
```

Please let me know if you'd like me to edit the commit. 
Appreciate your response.

me-no-dev commented:
Yes please :) revert your change and add the new one and we will merge it

me-no-dev commented:
The average frame rate looks a bit weird. Like 10 times lower than it should be... I wonder why

TNeutron commented:
Dear @me-no-dev 
I have reverted and added the change as you mentioned. 

Both FPS and Avg FPS is correct now. 
Here is the log camera stream log taken from this commit - 

```
14:51:03.753 -> [ 10931][I][app_httpd.cpp:290] stream_handler(): MJPG: 8132B 47ms (21.3fps), AVG: 32ms (31.2fps)
14:51:03.753 -> [ 10955][I][app_httpd.cpp:290] stream_handler(): MJPG: 6176B 24ms (41.7fps), AVG: 32ms (31.2fps)
14:51:03.785 -> [ 10988][I][app_httpd.cpp:290] stream_handler(): MJPG: 8132B 32ms (31.2fps), AVG: 33ms (30.3fps)
14:51:03.829 -> [ 11008][I][app_httpd.cpp:290] stream_handler(): MJPG: 6606B 19ms (52.6fps), AVG: 30ms (33.3fps)
14:51:03.862 -> [ 11064][I][app_httpd.cpp:290] stream_handler(): MJPG: 6606B 56ms (17.9fps), AVG: 32ms (31.2fps)
14:51:03.902 -> [ 11081][I][app_httpd.cpp:290] stream_handler(): MJPG: 6606B 16ms (62.5fps), AVG: 31ms (32.3fps)
14:51:03.902 -> [ 11112][I][app_httpd.cpp:290] stream_handler(): MJPG: 7081B 30ms (33.3fps), AVG: 31ms (32.3fps)
14:51:03.935 -> [ 11134][I][app_httpd.cpp:290] stream_handler(): MJPG: 8132B 22ms (45.5fps), AVG: 31ms (32.3fps)
14:51:03.981 -> [ 11163][I][app_httpd.cpp:290] stream_handler(): MJPG: 7074B 28ms (35.7fps), AVG: 31ms (32.3fps)
14:51:04.028 -> [ 11210][I][app_httpd.cpp:290] stream_handler(): MJPG: 7074B 47ms (21.3fps), AVG: 32ms (31.2fps)
14:51:04.072 -> [ 11282][I][app_httpd.cpp:290] stream_handler(): MJPG: 5928B 71ms (14.1fps), AVG: 34ms (29.4fps)
14:51:04.150 -> [ 11338][I][app_httpd.cpp:290] stream_handler(): MJPG: 6073B 56ms (17.9fps), AVG: 34ms (29.4fps)
14:51:04.191 -> [ 11400][I][app_httpd.cpp:290] stream_handler(): MJPG: 8132B 61ms (16.4fps), AVG: 36ms (27.8fps)
14:51:04.227 -> [ 11424][I][app_httpd.cpp:290] stream_handler(): MJPG: 6471B 24ms (41.7fps), AVG: 35ms (28.6fps)
14:51:04.274 -> [ 11455][I][app_httpd.cpp:290] stream_handler(): MJPG: 8132B 30ms (33.3fps), AVG: 34ms (29.4fps)
14:51:04.311 -> [ 11501][I][app_httpd.cpp:290] stream_handler(): MJPG: 8132B 45ms (22.2fps), AVG: 35ms (28.6fps)
14:51:04.311 -> [ 11521][I][app_httpd.cpp:290] stream_handler(): MJPG: 6769B 20ms (50.0fps), AVG: 34ms (29.4fps)
14:51:04.351 -> [ 11544][I][app_httpd.cpp:290] stream_handler(): MJPG: 8132B 22ms (45.5fps), AVG: 35ms (28.6fps)
14:51:04.383 -> [ 11576][I][app_httpd.cpp:290] stream_handler(): MJPG: 6473B 31ms (32.3fps), AVG: 35ms (28.6fps)
14:51:04.383 -> [ 11593][I][app_httpd.cpp:290] stream_handler(): MJPG: 7084B 16ms (62.5fps), AVG: 34ms (29.4fps)
14:51:04.419 -> [ 11628][I][app_httpd.cpp:290] stream_handler(): MJPG: 9427B 35ms (28.6fps), AVG: 34ms (29.4fps)
14:51:04.459 -> [ 11647][I][app_httpd.cpp:290] stream_handler(): MJPG: 8255B 19ms (52.6fps), AVG: 34ms (29.4fps)
14:51:04.505 -> [ 11681][I][app_httpd.cpp:290] stream_handler(): MJPG: 7861B 33ms (30.3fps), AVG: 34ms (29.4fps)
14:51:04.537 -> [ 11742][I][app_httpd.cpp:290] stream_handler(): MJPG: 7000B 60ms (16.7fps), AVG: 36ms (27.8fps)
14:51:04.570 -> [ 11774][I][app_httpd.cpp:290] stream_handler(): MJPG: 7000B 32ms (31.2fps), AVG: 34ms (29.4fps)
14:51:04.603 -> [ 11794][I][app_httpd.cpp:290] stream_handler(): MJPG: 7000B 20ms (50.0fps), AVG: 35ms (28.6fps)
14:51:04.603 -> [ 11810][I][app_httpd.cpp:290] stream_handler(): MJPG: 7861B 16ms (62.5fps), AVG: 34ms (29.4fps)
14:51:04.637 -> [ 11847][I][app_httpd.cpp:290] stream_handler(): MJPG: 6059B 36ms (27.8fps), AVG: 35ms (28.6fps)
14:51:04.673 -> [ 11866][I][app_httpd.cpp:290] stream_handler(): MJPG: 6533B 19ms (52.6fps), AVG: 34ms (29.4fps)
14:51:04.705 -> [ 11898][I][app_httpd.cpp:290] stream_handler(): MJPG: 6874B 31ms (32.3fps), AVG: 33ms (30.3fps)
14:51:04.705 -> [ 11914][I][app_httpd.cpp:290] stream_handler(): MJPG: 7060B 16ms (62.5fps), AVG: 31ms (32.3fps)
14:51:04.741 -> [ 11951][I][app_httpd.cpp:290] stream_handler(): MJPG: 7060B 36ms (27.8fps), AVG: 30ms (33.3fps)
14:51:04.781 -> [ 11971][I][app_httpd.cpp:290] stream_handler(): MJPG: 8641B 19ms (52.6fps), AVG: 28ms (35.7fps)
14:51:04.827 -> [ 12001][I][app_httpd.cpp:290] stream_handler(): MJPG: 7287B 30ms (33.3fps), AVG: 28ms (35.7fps)
14:51:04.871 -> [ 12060][I][app_httpd.cpp:290] stream_handler(): MJPG: 6910B 59ms (16.9fps), AVG: 29ms (34.5fps)
14:51:04.871 -> [ 12081][I][app_httpd.cpp:290] stream_handler(): MJPG: 6910B 20ms (50.0fps), AVG: 28ms (35.7fps)
14:51:04.904 -> [ 12103][I][app_httpd.cpp:290] stream_handler(): MJPG: 7468B 21ms (47.6fps), AVG: 28ms (35.7fps)
14:51:04.937 -> [ 12139][I][app_httpd.cpp:290] stream_handler(): MJPG: 7923B 36ms (27.8fps), AVG: 29ms (34.5fps)
14:51:05.012 -> [ 12195][I][app_httpd.cpp:290] stream_handler(): MJPG: 6339B 56ms (17.9fps), AVG: 30ms (33.3fps)
14:51:05.012 -> [ 12216][I][app_httpd.cpp:290] stream_handler(): MJPG: 5950B 20ms (50.0fps), AVG: 30ms (33.3fps)
14:51:05.089 -> [ 12298][I][app_httpd.cpp:290] stream_handler(): MJPG: 6681B 82ms (12.2fps), AVG: 33ms (30.3fps)
14:51:05.131 -> [ 12318][I][app_httpd.cpp:290] stream_handler(): MJPG: 7029B 19ms (52.6fps), AVG: 33ms (30.3fps)
14:51:05.209 -> [ 12389][I][app_httpd.cpp:290] stream_handler(): MJPG: 7029B 71ms (14.1fps), AVG: 34ms (29.4fps)
14:51:05.209 -> [ 12412][I][app_httpd.cpp:290] stream_handler(): MJPG: 7923B 22ms (45.5fps), AVG: 33ms (30.3fps)
14:51:05.245 -> [ 12437][I][app_httpd.cpp:290] stream_handler(): MJPG: 6174B 25ms (40.0fps), AVG: 32ms (31.2fps)
14:51:05.245 -> [ 12454][I][app_httpd.cpp:290] stream_handler(): MJPG: 6791B 17ms (58.8fps), AVG: 32ms (31.2fps)
14:51:05.277 -> [ 12481][I][app_httpd.cpp:290] stream_handler(): MJPG: 6015B 26ms (38.5fps), AVG: 33ms (30.3fps)
14:51:05.313 -> [ 12523][I][app_httpd.cpp:290] stream_handler(): MJPG: 6928B 41ms (24.4fps), AVG: 33ms (30.3fps)
14:51:05.347 -> [ 12556][I][app_httpd.cpp:290] stream_handler(): MJPG: 9079B 33ms (30.3fps), AVG: 34ms (29.4fps)
14:51:05.392 -> [ 12579][I][app_httpd.cpp:290] stream_handler(): MJPG: 7613B 22ms (45.5fps), AVG: 33ms (30.3fps)
14:51:05.424 -> [ 12603][I][app_httpd.cpp:290] stream_handler(): MJPG: 7613B 24ms (41.7fps), AVG: 33ms (30.3fps)
14:51:05.424 -> [ 12626][I][app_httpd.cpp:290] stream_handler(): MJPG: 8491B 22ms (45.5fps), AVG: 33ms (30.3fps)
14:51:05.461 -> [ 12653][I][app_httpd.cpp:290] stream_handler(): MJPG: 7453B 26ms (38.5fps), AVG: 33ms (30.3fps)
14:51:05.461 -> [ 12670][I][app_httpd.cpp:290] stream_handler(): MJPG: 6488B 17ms (58.8fps), AVG: 32ms (31.2fps)
14:51:05.498 -> [ 12690][I][app_httpd.cpp:290] stream_handler(): MJPG: 7441B 19ms (52.6fps), AVG: 30ms (33.3fps)
14:51:05.498 -> [ 12707][I][app_httpd.cpp:290] stream_handler(): MJPG: 7029B 17ms (58.8fps), AVG: 30ms (33.3fps)
14:51:05.577 -> [ 12757][I][app_httpd.cpp:290] stream_handler(): MJPG: 7029B 49ms (20.4fps), AVG: 32ms (31.2fps)
14:51:05.622 -> [ 12788][I][app_httpd.cpp:290] stream_handler(): MJPG: 8170B 31ms (32.3fps), AVG: 31ms (32.3fps)
14:51:05.669 -> [ 12833][I][app_httpd.cpp:290] stream_handler(): MJPG: 6402B 45ms (22.2fps), AVG: 31ms (32.3fps)
```

I'm still learning my way around git. Again, I appreciate your patience and kind response :) Looking forward to the merge 😅
